profile.cc

```diff
@@ -1,5 +1,46 @@
 #include "profile.h"
+#include <cmath>
+#include <iostream>
 
-const double Profile::GetTime() {
-  return 0.0;
-}
+const double Profile::GetTime(bool ignore_max_velocity, ProfilePoint current,
+                              ProfilePoint goal) {
+  double time;
+  double time_accelerating;
+  double time_at_constant_velocity;
+  double time_decelerating;
+
+  double distance_traveled_while_accelerating;
+  double distance_traveled_at_constant_velocity;
+  double distance_traveled_while_decelerating;
+
+  double position_initial = current.position;
+  double position_goal = goal.position;
+  double velocity_initial = current.velocity;
+  double velocity_final = kMaxVelocity;
+  double velocity_goal = goal.velocity;
+  double acceleration = kMaxAcceleration;
+
+  if (ignore_max_velocity) {  // first case
+    time = (-velocity_initial +
+            std::sqrt(std::pow(velocity_initial, 2) +
+                      (4 * .5 * acceleration * goal.position))) /
+           acceleration;
+  } else {  // second case
+    time_accelerating =
+        std::abs(velocity_final - velocity_initial) / acceleration;
+    distance_traveled_while_accelerating =
+        (.5 * std::abs(velocity_final + velocity_initial) * time_accelerating);
+    time_decelerating = std::abs(velocity_goal - velocity_final) / acceleration;
+    distance_traveled_while_decelerating =
+        (.5 * std::abs(velocity_goal - velocity_final) * time_decelerating);
+    distance_traveled_at_constant_velocity =
+        position_goal - distance_traveled_while_accelerating -
+        distance_traveled_while_decelerating;
+    time_at_constant_velocity =
+        distance_traveled_at_constant_velocity / velocity_final;
```

Comment:
what if it never gets to constant velocity

Reply:
sucks to suck